### ISSUE TITLE
["BUG"] x-cmd installation creates .bash_profile in $HOME, which blocks .bashrc from loading | [mod=boot] [v0.4.14]

### ISSUE THREAD

**Issue by JamesDBartlett3** [x-cmd/x-cmd#82]
**Describe the bug**
When the x-cmd installer runs, it automatically creates a file called `.bash_profile` in the user's home directory (typically `/home/username`) and adds the following content:
```bash
[ ! -f "$HOME/.x-cmd.root/X" ] || . "$HOME/.x-cmd.root/X" # boot up x-cmd.
```
The [default behavior of Bash](https://www.reddit.com/r/bashonubuntuonwindows/comments/l6y66i/comment/gl3kvci/) is to source a file called `~/.bashrc` on launch, ***unless a file called `~/.bash_profile` exists***, in which case it will skip `.bashrc` and source `.bash_profile` instead.

So, in any system that has been setup with `~/.bashrc` as the default location for the user's shell settings and preferences, installing x-cmd effectively blocks those settings and preferences from being loaded into the Bash session the next time the user logs in.

**System (please complete the following information):**
 - Version: 
> .c7d76596
> version: v0.4.10
> date: "2024-09-10 11:33:50+0000"
> sum: c7d765961ce18be028f3095c5df96b294323f46b2fe58001a4722cd418f8baddbc5ef0a10e553cbfd7815ff99ab5fdeb052f29552998d64373144f9d8bfa6c88
 - Shell: Bash
 - OS: Tested in Raspbian OS and WSL2 Ubuntu
 - Arch: amd64

**Screenshots**
- None

**To Reproduce**
1. Install x-cmd on a system where `~/.bashrc` is the default location for the user's shell settings and preferences, and `~/.bash_profile` does not exist by default (e.g. Raspbian OS, WSL2 Ubuntu, and possibly others).
2. Log out.
3. Log back in.

**Expected behavior**
The x-cmd installer should first check to see whether `~/.bash_profile` exists, and only insert its code if the file does already exist. Otherwise, it should insert the code into `~/.bashrc`

**Additional context**
- None

**Comment by edwinjhlee** [comment]
Thank you for reporting this issue. I was unaware of this behavior.
    
I am a mac user in recent years when developing x-cmd. On macOS, Terminal.app ( the system default terminal) sources `.bash_profile` because it treats new terminal windows as login shells. Other terminals typically source `.bashrc`. That's why we have to inject the boot up line in the bash_profile and bashrc. ( I don't like this, but it makes x-cmd's installation friendly to users. )

---

Currently, I prefer to leave this configuration up to the user. For example, you can add the following line to your `.bash_profile`: `[ -f ~/.bashrc ] && . ~/.bashrc`

    
It would be quite difficult for x-cmd to handle this configuration universally in a way that works seamlessly for all users and environments.

**Comment by edwinjhlee** [comment]
Today I revisiting this issue,  I am consider introducing a strategy like this:

Always add boot line into the bashrc.
Add boot line into the ~/.bash_profile if and only if this file is detected.

**Comment by ivan94fi** [comment]
I am also seeing this behavior.

I acknowledged this bug when suddenly the servers where I installed x-cmd started not loading `.bashrc` upon ssh connection. Before `.bashrc` was loaded also when starting ssh because it is sourced inside `.profile` (at least in Ubuntu), but the `.bash_profile` created by x-cmd shadows the `.profile`, and it is ignored.

Also, I noticed that in both the servers where x-cmd is installed, that the `.bash_history` file was overwritten: maybe this is because I started working without a loaded `.bashrc` that defines the history related configuration. One should not rely on the shell history to save important commands, but the history feature is very useful nonetheless and it is a shame when it is deleted by accident.

So I suggest that the strategy you outlined in the previous comment is implemented as soon as possible.

Thank you!

**Comment by ivan94fi** [comment]
Thank you! I will try it as soon as possible

**Comment by JamesDBartlett3** [comment]
> We've just released v0.4.14. Please upgrade to this latest version and see if the issue is resolved.
> 
> Thank you for suggestions ~
> 
> @JamesDBartlett3 @ivan94fi

I upgraded to the latest version yesterday, and everything seems to be working fine now. Thank you!
